docs/versioned_docs/version-zenith/guides/620942-github-google-cloud.mdx

```diff
@@ -0,0 +1,319 @@
+      - It accepts one argument: a `Directory` representing the directory containing the application source code.
+      - It uses the client's `Container().from()` method to initialize a new container from a base image. In this example, the base image is the `node:21` image. This method returns a `Container` representing an OCI-compatible container image.
+      - It uses the `Container.withDirectory()` method to return the container image with the application source code written at the `/home/node` path, and the `Container.withWorkdir()` method to set the working directory in the container.
+      - It chains the `Container.withExec()` method again to install dependencies with `npm install` and sets the container entrypoint using the `Container.withEntrypoint()` method.
+
+    The `publish()` function takes care of publishing the container image to Google Artifact Registry.
+      - It accepts five arguments: a `Directory` representing the directory containing the application source code, and strings representing the Google Cloud project name, Google Artifact Registry location and Google Artifact Registry repository name, and a `Secret` representing Google Cloud credentials.
+      - It invokes the `build()` function to produce a `Container` representing the application image.
+      - It uses the `Container.publish()` method to publish the container to the Google Artifact Registry repository and returns the SHA identifier of the published image.
+
+    The `deploy()` function takes care of deploying the application to an existing Google Cloud Run service.
+      - It accepts seven arguments: a `Directory` representing the directory containing the application source code, and strings representing the Google Cloud project name, Google Artifact Registry location, Google Artifact Registry repository name, Google Cloud Run service location, Google Cloud Run service name, and a `Secret` representing Google Cloud credentials.
+      - It invokes the `publish()` function to build and publish a `Container` representing the application image.
+      - It uses the Google Cloud Run module to deploy the container image to the named Google Cloud Run service.
+
+</TabItem>
+</Tabs>
+
+## Step 6: Test the Dagger pipeline on the local host
+
+Once credentials are configured, test the Dagger Functions by running the command below from the application directory. Remember to update the `registry-location`, `repository`, `service`, and `service-location` arguments if you specified different values when creating the various Google Cloud services in Step 3 and Step 4, and update the `SERVICE-ACCOUNT-KEY-FILE` placeholder with the local filesystem path to your JSON service key file from Step 1.
+
+```shell
+dagger call deploy --source . --project myproject --registry-location us-central1 --repository myregistry/myapp --service-location us-central1 --service myservice  --credential file:SERVICE-ACCOUNT-KEY-FILE
```

Comment:
@sipsma updated this per your earlier comment (which is no longer visible since I deleted that section of the text)